src/math/PercentageMath.sol

```diff
@@ -112,29 +112,23 @@ library PercentageMath {
         // Must revert if
         //     percentage > PERCENTAGE_FACTOR
         // or if
-        //     x1 = x * (PERCENTAGE_FACTOR - percentage) + HALF_PERCENTAGE_FACTOR > type(uint256).max
-        //     <=> (PERCENTAGE_FACTOR - percentage) > 0 and x > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)
-        // or if
-        //     x2 = y * percentage + HALF_PERCENTAGE_FACTOR > type(uint256).max
-        //     <=> y * percentage > type(uint256).max - HALF_PERCENTAGE_FACTOR
+        //     y * percentage + HALF_PERCENTAGE_FACTOR > type(uint256).max
         //     <=> percentage > 0 and y > (type(uint256).max - HALF_PERCENTAGE_FACTOR) / percentage
-        //
-        // Note: x1 / PERCENTAGE_FACTOR + x2 / PERCENTAGE_FACTOR <= x * (1 - p) + 1 + y * p + 1 where p is the fractional number percentage / PERCENTAGE_FACTOR
-        // <= max(x, y) + 2 <= type(uint256).max - HALF_PERCENTAGE_FACTOR + 2 <= type(uint256).max
+        // or if
+        //     x * (PERCENTAGE_FACTOR - percentage) + y * percentage + HALF_PERCENTAGE_FACTOR > type(uint256).max
+        //     <=> (PERCENTAGE_FACTOR - percentage) and (x > type(uint256).max - HALF_PERCENTAGE_FACTOR - y * percentage ) / (PERCENTAGE_FACTOR - percentage)
         assembly {
             z := sub(PERCENTAGE_FACTOR, percentage) // Temporary assignment to save gas.
-
             if or(
-                or(gt(percentage, PERCENTAGE_FACTOR), mul(z, gt(x, div(MAX_UINT256_MINUS_HALF_PERCENTAGE, z)))),
-                mul(percentage, gt(y, div(MAX_UINT256_MINUS_HALF_PERCENTAGE, percentage)))
+                gt(percentage, PERCENTAGE_FACTOR),
+                or(
+                    mul(percentage, gt(y, div(MAX_UINT256_MINUS_HALF_PERCENTAGE, percentage))),
+                    mul(z, gt(x, div(sub(sub(MAX_UINT256, HALF_PERCENTAGE_FACTOR), mul(y, percentage)), z)))
```

Comment:
Maybe chose the same thing for the 2 lines, either `MAX_UINT256_MINUS_HALF_PERCENTAGE` or `sub(MAX_UINT256, HALF_PERCENTAGE_FACTOR)`